spec/jobs/abstract_delivery_job_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe AbstractDeliveryJob, type: :job do
+  # NOTE: Test a concrete impl of the abstract job so the call to `#bucket` does not raise
+  subject(:job_implementation) do
+    Class.new(described_class) do
+      def bucket
+        Struct.new(:unused) do
+          def object(arg); end
+        end.new
+      end
+    end
+  end
```

Comment:
I moved the testing out of the S3 West job and split the tests into the abstract job and the new service, @jmartin-sul @ndushay